src/Console/CommandRunner.php

```diff
@@ -368,6 +369,14 @@ protected function runShell(Shell $shell, array $argv)
      */
     protected function createCommand(string $className, ConsoleIo $io)
     {
+        if (!$this->factory) {
```

Comment:
Static analysers will be happy if this is moved into the constructor.

Reply:
Hrm. Doing that in the constructor would result in the container being built before the application's bootstrap was called which will not work great.